modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/schema/builder/SchemaBuilder.java

```diff
@@ -747,6 +777,11 @@ private boolean shouldGeneratePojoChildElements(Class<?> type)
             {
                 return IntrospectionUtils.isInstantiable(type) && !getExposedFields(type).isEmpty();
             }
+
+            private boolean shouldForceOptional()
+            {
+                return !(parameterModel.isRequired() && ExpressionSupport.REQUIRED.equals(parameterModel.getExpressionSupport()));
```

Comment:
use == to compare enums. Also, negating the whole expressions seems to make the statement unnecessarily hard to read. It looks like this could be written in a simpler way